hanson_at_fb.py

```diff
@@ -0,0 +1,33 @@
+# given a list of strings, determine which are anagrams and group them together.
+# return a list of lists of anagrams
+
+# input ['cat', 'act', 'dog', 'god', 'def']
+# output [['cat', 'act'], ['dog', 'god'], ['def']]
+
+# lists are assigned by reference, so list1 = list2, list 1 points to list 2
+# to clone a list, you want to do list1 = list(list2)
+
+# sorted(string) = list of chars of string
+# sort(list) = modifies list in-place
+# ''.join(list), joins each element of the list together with '' (no space) in between
+
+
+def find_anagrams (some_list):
+	grouped_words = {}
+	for word in some_list:
+		sorted_word = ''.join(sorted(word))
+		print sorted_word
+		#if grouped_words[sorted_word]:
+		if sorted_word in grouped_words:
```

Comment:
This is a common enough pattern that Python has something built-in for this! `defaultdict`[0] let's you assign a default for every key, even if they don't exist yet. So no need for this if statement!

[0] https://docs.python.org/2/library/collections.html#defaultdict-examples